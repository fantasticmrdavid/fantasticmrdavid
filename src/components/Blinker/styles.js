import styled, { css } from 'styled-components';
import { blink } from 'styles/animations';
import theme from 'styles/theme';

export const Cursor = styled.div`
  position: relative;
  display: inline-block;
  width: 0.5rem;
  background-color: ${theme.colorFont};
  margin-left: 0.15rem;
  animation: ${css`${blink} 1s step-end infinite`};
  border-bottom: 2px solid ${theme.colorFont};
`;
