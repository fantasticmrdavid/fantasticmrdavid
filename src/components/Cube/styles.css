.Cube_container {
  position: absolute;
  bottom: 0px;
  right: 0px;
	perspective: 800px;
	perspective-origin: 50% 150px;
  transition: 0.5s;
  transform: scale(2);

  &.loading {
    transform: scale(8);
  }
}

.Cube {
	position: absolute;;
  bottom: 225px;
  right: -100px;
	width: 300px;
	transform-style: preserve-3d;
  animation: spin 0.6s normal linear;
  animation-fill-mode: forwards;

  div {
  	position: absolute;
  	width: 300px;
  	height: 300px;
    background: rgba(209, 43, 43, 0.1);
    border: 1px solid rgba(209, 43, 43, 0.2);
  }

  .back {
  	transform: translateZ(-150px) rotateY(180deg);
  }
  .right {
  	transform: rotateY(-270deg) translateX(150px);
  	transform-origin: top right;
  }
  .left {
  	transform: rotateY(270deg) translateX(-150px);
  	transform-origin: center left;
  }
  .top {
  	transform: rotateX(-90deg) translateY(-150px);
  	transform-origin: top center;
  }
  .bottom {
  	transform: rotateX(90deg) translateY(150px);
  	transform-origin: bottom center;
  }
  .front {
  	transform: translateZ(150px);
  }
}

@keyframes spin {
	from { transform: rotateX(15deg) rotateY(15deg); }
	to { transform: rotateX(-60deg) rotateY(60deg); }
}
