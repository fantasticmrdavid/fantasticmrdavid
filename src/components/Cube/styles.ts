import styled, { css, keyframes } from 'styled-components';
import { transparentize } from 'polished';

const cubeColors = [
  'rgb(209, 43, 43)',
  'rgb(66, 33, 99)',
  'rgb(0, 0, 255)',
];

interface ContainerProps {
  loading: boolean,
}

const spin = keyframes`
  from { transform: rotateX(15deg) rotateY(15deg); }
  to { transform: rotateX(-60deg) rotateY(60deg); }
`;

const cycleColor = keyframes`
  ${cubeColors.map((c: string, i: number) => `
    ${i * (100 / cubeColors.length)}% {
      background-color: ${transparentize(0.9, c)};
      border: ${`1px solid ${transparentize(0.8, c)}`};
    }`)}
    100% {
      background-color: ${transparentize(0.9, cubeColors[0])};
      border: ${`1px solid ${transparentize(0.8, cubeColors[0])}`};
    }
`;

export const Container = styled.div`
  position: absolute;
  bottom: 0px;
  right: 0px;
  perspective: 800px;
  perspective-origin: 50% 150px;
  transition: 0.5s;
  transform: ${css`scale(${(props: ContainerProps) => (props.loading ? 8 : 2)})`};
`;

export const Polygon = styled.div`
  position: absolute;
  bottom: 225px;
  right: -100px;
  width: 300px;
  transform-style: preserve-3d;
  animation: ${css`${spin} 0.6s normal linear`};
  animation-fill-mode: forwards;
`;

const Side = styled.div`
  position: absolute;
  width: 300px;
  height: 300px;
  animation: ${css`${cycleColor} 45s normal infinite`};
`;

export const Back = styled(Side)`
  transform: translateZ(-150px) rotateY(180deg);
`;

export const Right = styled(Side)`
  transform: rotateY(-270deg) translateX(150px);
  transform-origin: top right;
`;

export const Left = styled(Side)`
  transform: rotateY(270deg) translateX(-150px);
  transform-origin: center left;
`;

export const Top = styled(Side)`
  transform: rotateX(-90deg) translateY(-152px);
  transform-origin: top center;
`;

export const Bottom = styled(Side)`
  transform: rotateX(90deg) translateY(150px);
  transform-origin: bottom center;
`;

export const Front = styled(Side)`
  transform: translateZ(150px);
`;
