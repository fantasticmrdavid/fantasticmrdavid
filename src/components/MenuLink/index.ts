import { withRouter } from 'react-router-dom';
import Ui from './ui';

export default withRouter(Ui);
