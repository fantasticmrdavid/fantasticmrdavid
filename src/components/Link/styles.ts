import styled from 'styled-components';
import theme from 'styles/theme';
import { media } from 'styles/utils';

export const Link = styled.a`
  color: ${theme.colorWhite};
  text-decoration: none;
  border-bottom: ${`1px solid ${theme.colorWhite}`};
  transition: 0.3s;

  ${media.medium`
    &:hover {
      border-bottom-color: ${theme.colorRed};
    }
  `};
`;
