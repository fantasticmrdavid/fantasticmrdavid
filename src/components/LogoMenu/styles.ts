import styled from 'styled-components';
import theme from 'styles/theme';
import { media } from 'styles/utils';

interface ContainerProps {
  isOpen: boolean,
}

interface CtaProps {
  isOpen: boolean,
}

interface NavProps {
  isOpen: boolean,
}

interface SubtitleProps {
  isOpen: boolean,
}

interface CogIconWrapperProps {
  isOpen?: boolean,
}

export const Container = styled.nav<ContainerProps>`
  position: relative;
  font-family: ${theme.fontHeading};
  display: flex;
  flex-wrap: wrap;
  align-content: center;
  cursor: pointer;
  min-height: 0px;
  padding: 1.5em 0;
  padding-bottom: ${(props) => (props.isOpen ? '0px' : undefined)};

  ${media.small`
    min-height: 100%;
    padding: 0;
  `};
`;

export const Title = styled.h1`
  padding: 0 4rem 0 2rem;
  font-size: 1.8rem;

  ${media.small`
    font-size: 2rem;
  `};
`;

export const Cta = styled.a<CtaProps>`
  position: absolute;
  display: ${(props) => (props.isOpen ? 'none' : 'flex')};
  justify-content: center;
  align-items: center;
  top: calc(50% - 25px);
  right: 2rem;
  border-radius: 50%;
  border: ${`2px solid ${theme.colorDarkGrey}`};
  font-size: 2rem;
  color: ${theme.colorDarkGrey};
  width: 50px;
  height: 50px;
  z-index: ${(props) => (props.isOpen ? -1 : 1)};
  opacity: ${(props) => (props.isOpen ? 'undefined' : 1)};
  transition: 0.3s;

  ${media.small`
    display: flex;
  `};

  ${media.hover`
    &:hover {
      background: ${theme.colorCharcoal};
    }
  `};
`;

export const Nav = styled.nav<NavProps>`
  flex-wrap: wrap;
  font-size: 1.5rem;
  width: 100%;
  margin: ${(props) => (props.isOpen ? '0.5rem 0' : undefined)};
  max-height: ${(props) => (props.isOpen ? '1000px' : '0px')};
  transition: 0.5s;
  opacity: ${(props) => (props.isOpen ? 1 : 0)};
  overflow: hidden;
  background-color: #444;
  border-top: ${(props) => (props.isOpen ? '1px solid #333' : undefined)};
  border-bottom: ${(props) => (props.isOpen ? '1px solid #333' : undefined)};
`;

export const Subtitle = styled.div<SubtitleProps>`
  display: ${(props) => (props.isOpen ? 'none' : 'flex')};
  position: relative;
  font-size: 1rem;
  width: 100%;
  padding: 0 2rem;
  margin-top: 0.25em;
  
  h2 {
    display: flex;
    align-items: center;
  }

  ${media.small`
    display: flex;
  `};
`;

export const CogIconWrapper = styled.div<CogIconWrapperProps>`
  position: relative;
  transition: 0.6s;
  transform: ${(props) => (props.isOpen ? 'rotate(150deg)' : undefined)};
  left: -0.25em;
  opacity: 0.5;
`;
