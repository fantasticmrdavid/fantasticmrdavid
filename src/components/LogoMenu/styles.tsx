import React from 'react';
import styled from 'styled-components';
import theme from 'styles/theme';
import { media } from 'styles/utils';
import IconComponent, { Props as IconComponentProps } from 'components/Icon';

interface ContainerProps {
  active: boolean,
}

interface CtaProps {
  active: boolean,
}

interface NavProps {
  active: boolean,
}

interface SubtitleProps {
  active: boolean,
}

interface IconProps {
  active?: boolean,
}

export const Container = styled.nav<ContainerProps>`
  position: relative;
  font-family: ${theme.fontHeading};
  display: flex;
  flex-wrap: wrap;
  align-content: center;
  cursor: pointer;
  min-height: 0px;
  padding: 1.5em 0;
  padding-bottom: ${props => (props.active ? '0px' : undefined)};

  ${media.small`
    min-height: 100%;
    padding: 0;
  `};
`;

export const Title = styled.h1`
  padding: 0 4rem 0 2rem;
  font-size: 1.8rem;

  ${media.small`
    font-size: 2rem;
  `};
`;

export const Cta = styled.a<CtaProps>`
  position: absolute;
  display: ${props => (props.active ? 'none' : 'flex')};
  justify-content: center;
  align-items: center;
  top: calc(50% - 25px);
  right: 2rem;
  border-radius: 50%;
  border: ${`2px solid ${theme.colorDarkGrey}`};
  font-size: 2rem;
  color: ${theme.colorDarkGrey};
  width: 50px;
  height: 50px;
  z-index: ${props => (props.active ? -1 : 1)};
  opacity: ${props => (props.active ? 'undefined' : 1)};
  transition: 0.3s;

  ${media.small`
    display: flex;
  `};

  ${media.hover`
    &:hover {
      background: ${theme.colorCharcoal};
    }
  `};
`;

export const Nav = styled.nav<NavProps>`
  flex-wrap: wrap;
  font-size: 1.5rem;
  width: 100%;
  margin: ${props => (props.active ? '0.5rem 0' : undefined)};
  max-height: ${props => (props.active ? '1000px' : '0px')};
  transition: 0.5s;
  opacity: ${props => (props.active ? 1 : 0)};
  overflow: hidden;
  background-color: #444;
  border-top: ${props => (props.active ? '1px solid #333' : undefined)};
  border-bottom: ${props => (props.active ? '1px solid #333' : undefined)};
`;

export const Subtitle = styled.div<SubtitleProps>`
  display: ${props => (props.active ? 'none' : 'flex')};
  position: relative;
  font-size: 1rem;
  width: 100%;
  padding: 0 2rem;
  margin-top: 0.25em;

  ${media.small`
    display: flex;
  `};
`;

export const Icon = styled(({ active, ...rest }: IconProps & IconComponentProps) => <IconComponent {...rest} />)`
  position: relative;
  transition: 0.6s;
  transform: ${props => (props.active ? 'rotate(150deg)' : undefined)};
`;

export const CogIcon = styled(Icon)`
  left: -0.25em;
  opacity: 0.5;
`;
