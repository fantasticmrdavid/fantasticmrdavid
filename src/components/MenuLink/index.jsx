import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import LocationContent from './LocationContent';
import styles from './styles.css';

class MenuLink extends PureComponent {
  constructor() {
    super();
    this.boundHandleNavClick = this.handleNavClick.bind(this);
  }

  handleNavClick(e) {
    const { target, current, updateLocation } = this.props;
    e.preventDefault();
    e.stopPropagation();
    updateLocation(current ? 'home' : target);
  }

  render() {
    const { label, target, current, menuActive } = this.props;
    return (
      <div className={`${styles.MenuLink} ${current && styles.current} ${menuActive && styles.menuActive}`} onClick={this.boundHandleNavClick}>
        <div className={`${styles.label} ${styles[target]}`} data-target={target} onClick={this.boundHandleNavClick}>{ label }</div>
        <LocationContent location={target} current={current} />
      </div>
    );
  }
}

MenuLink.propTypes = {
  updateLocation: PropTypes.func.isRequired,
  target: PropTypes.string.isRequired,
  current: PropTypes.bool,
  menuActive: PropTypes.bool,
  label: PropTypes.string,
};

export default MenuLink;
