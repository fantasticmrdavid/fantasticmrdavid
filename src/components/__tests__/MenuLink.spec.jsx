import React from 'react';
import { mount } from 'enzyme';
import Ui from 'components/MenuLink/ui';
import { Label } from 'components/MenuLink/styles';

const props = {
  target: 'contact',
  current: false,
  menuActive: true,
  label: 'Contact',
  updateLocation: jest.fn(),
  history: {
    push: jest.fn(),
  },
};

describe('Component -> MenuLink', () => {
  it('should render the "contact" MenuLink UI without throwing an error', () => {
    expect(mount(<Ui {...props} />)).toMatchSnapshot();
  });

  it('should render the "work" MenuLink UI without throwing an error', () => {
    expect(mount(<Ui {...props} target="work" />)).toMatchSnapshot();
  });

  it('should render the "about" MenuLink UI without throwing an error', () => {
    expect(mount(<Ui {...props} target="about" />)).toMatchSnapshot();
  });

  it('should call updateLocation and push history when clicked', () => {
    const component = mount(<Ui {...props} />);
    component.find(Label).simulate('click');
    expect(props.updateLocation).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalled();
  });
});
