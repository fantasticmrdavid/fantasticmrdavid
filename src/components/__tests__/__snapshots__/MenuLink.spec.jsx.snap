// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> MenuLink should render the "about" MenuLink UI without throwing an error 1`] = `
<Memo()
  current={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  label="Contact"
  menuActive={true}
  target="about"
  updateLocation={[MockFunction]}
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-sc-1w8m3j0-0",
            "isStatic": false,
            "lastClassName": "dmrGVg",
            "rules": Array [
              "display:block;position:relative;color:inherit;&.current{transition:0s;}",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-sc-1w8m3j0-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="styles__Container-sc-1w8m3j0-0 dmrGVg"
      >
        <styles__Label
          current={false}
          menuActive={true}
          onClick={[Function]}
          target="about"
        >
          <StyledComponent
            current={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Label-sc-1w8m3j0-1",
                  "isStatic": false,
                  "lastClassName": "dcJTwn",
                  "rules": Array [
                    "position:relative;padding:",
                    [Function],
                    ";transition:0.5s;color:",
                    [Function],
                    ";font-size:",
                    [Function],
                    ";font-weight:",
                    [Function],
                    ";cursor:pointer;background-color:",
                    [Function],
                    ";&:before{position:absolute;top:0px;left:0px;content:\\"\\";height:100%;transition:0.3s;background-color:",
                    "#555",
                    ";}",
                    "@media (hover:hover){",
                    "
    &:hover {
      padding-left: ",
                    [Function],
                    ";
      background-color: ",
                    [Function],
                    ";

      &:before {
        width: ",
                    [Function],
                    ";
        background-color: ",
                    [Function],
                    ";
      }
    }
  ",
                    "}",
                    ";",
                  ],
                },
                "displayName": "styles__Label",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Label-sc-1w8m3j0-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            menuActive={true}
            onClick={[Function]}
            target="about"
          >
            <div
              className="styles__Label-sc-1w8m3j0-1 dcJTwn"
              onClick={[Function]}
              target="about"
            >
              Contact
            </div>
          </StyledComponent>
        </styles__Label>
        <Memo()
          current={false}
          location="about"
        >
          <styles__Content
            current={false}
          >
            <StyledComponent
              current={false}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Content-xptth5-0",
                    "isStatic": false,
                    "lastClassName": "iheHWk",
                    "rules": Array [
                      "display:none;font-size:1.1rem;font-weight:300;line-height:1.3;max-height:",
                      [Function],
                      ";padding:",
                      [Function],
                      ";color:",
                      [Function],
                      ";background-color:",
                      [Function],
                      ";overflow:hidden;transition:0.6s;",
                      "@media (min-width:",
                      "35.5",
                      "em){",
                      "
    display: block;
  ",
                      "}",
                      ";",
                    ],
                  },
                  "displayName": "styles__Content",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Content-xptth5-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="styles__Content-xptth5-0 iheHWk"
              >
                <_default>
                  <div>
                    Meet the captain of the ship.
                  </div>
                </_default>
              </div>
            </StyledComponent>
          </styles__Content>
        </Memo()>
      </div>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;

exports[`Component -> MenuLink should render the "contact" MenuLink UI without throwing an error 1`] = `
<Memo()
  current={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  label="Contact"
  menuActive={true}
  target="contact"
  updateLocation={[MockFunction]}
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-sc-1w8m3j0-0",
            "isStatic": false,
            "lastClassName": "dmrGVg",
            "rules": Array [
              "display:block;position:relative;color:inherit;&.current{transition:0s;}",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-sc-1w8m3j0-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="styles__Container-sc-1w8m3j0-0 dmrGVg"
      >
        <styles__Label
          current={false}
          menuActive={true}
          onClick={[Function]}
          target="contact"
        >
          <StyledComponent
            current={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Label-sc-1w8m3j0-1",
                  "isStatic": false,
                  "lastClassName": "eIGLXT",
                  "rules": Array [
                    "position:relative;padding:",
                    [Function],
                    ";transition:0.5s;color:",
                    [Function],
                    ";font-size:",
                    [Function],
                    ";font-weight:",
                    [Function],
                    ";cursor:pointer;background-color:",
                    [Function],
                    ";&:before{position:absolute;top:0px;left:0px;content:\\"\\";height:100%;transition:0.3s;background-color:",
                    "#555",
                    ";}",
                    "@media (hover:hover){",
                    "
    &:hover {
      padding-left: ",
                    [Function],
                    ";
      background-color: ",
                    [Function],
                    ";

      &:before {
        width: ",
                    [Function],
                    ";
        background-color: ",
                    [Function],
                    ";
      }
    }
  ",
                    "}",
                    ";",
                  ],
                },
                "displayName": "styles__Label",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Label-sc-1w8m3j0-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            menuActive={true}
            onClick={[Function]}
            target="contact"
          >
            <div
              className="styles__Label-sc-1w8m3j0-1 eIGLXT"
              onClick={[Function]}
              target="contact"
            >
              Contact
            </div>
          </StyledComponent>
        </styles__Label>
        <Memo()
          current={false}
          location="contact"
        >
          <styles__Content
            current={false}
          >
            <StyledComponent
              current={false}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Content-xptth5-0",
                    "isStatic": false,
                    "lastClassName": "iheHWk",
                    "rules": Array [
                      "display:none;font-size:1.1rem;font-weight:300;line-height:1.3;max-height:",
                      [Function],
                      ";padding:",
                      [Function],
                      ";color:",
                      [Function],
                      ";background-color:",
                      [Function],
                      ";overflow:hidden;transition:0.6s;",
                      "@media (min-width:",
                      "35.5",
                      "em){",
                      "
    display: block;
  ",
                      "}",
                      ";",
                    ],
                  },
                  "displayName": "styles__Content",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Content-xptth5-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="styles__Content-xptth5-0 iheHWk"
              >
                <_default>
                  <div>
                    For relaxing times, make it Suntory time.
                  </div>
                </_default>
              </div>
            </StyledComponent>
          </styles__Content>
        </Memo()>
      </div>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;

exports[`Component -> MenuLink should render the "work" MenuLink UI without throwing an error 1`] = `
<Memo()
  current={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  label="Contact"
  menuActive={true}
  target="work"
  updateLocation={[MockFunction]}
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-sc-1w8m3j0-0",
            "isStatic": false,
            "lastClassName": "dmrGVg",
            "rules": Array [
              "display:block;position:relative;color:inherit;&.current{transition:0s;}",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-sc-1w8m3j0-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="styles__Container-sc-1w8m3j0-0 dmrGVg"
      >
        <styles__Label
          current={false}
          menuActive={true}
          onClick={[Function]}
          target="work"
        >
          <StyledComponent
            current={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Label-sc-1w8m3j0-1",
                  "isStatic": false,
                  "lastClassName": "SLhrz",
                  "rules": Array [
                    "position:relative;padding:",
                    [Function],
                    ";transition:0.5s;color:",
                    [Function],
                    ";font-size:",
                    [Function],
                    ";font-weight:",
                    [Function],
                    ";cursor:pointer;background-color:",
                    [Function],
                    ";&:before{position:absolute;top:0px;left:0px;content:\\"\\";height:100%;transition:0.3s;background-color:",
                    "#555",
                    ";}",
                    "@media (hover:hover){",
                    "
    &:hover {
      padding-left: ",
                    [Function],
                    ";
      background-color: ",
                    [Function],
                    ";

      &:before {
        width: ",
                    [Function],
                    ";
        background-color: ",
                    [Function],
                    ";
      }
    }
  ",
                    "}",
                    ";",
                  ],
                },
                "displayName": "styles__Label",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Label-sc-1w8m3j0-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            menuActive={true}
            onClick={[Function]}
            target="work"
          >
            <div
              className="styles__Label-sc-1w8m3j0-1 SLhrz"
              onClick={[Function]}
              target="work"
            >
              Contact
            </div>
          </StyledComponent>
        </styles__Label>
        <Memo()
          current={false}
          location="work"
        >
          <styles__Content
            current={false}
          >
            <StyledComponent
              current={false}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Content-xptth5-0",
                    "isStatic": false,
                    "lastClassName": "iheHWk",
                    "rules": Array [
                      "display:none;font-size:1.1rem;font-weight:300;line-height:1.3;max-height:",
                      [Function],
                      ";padding:",
                      [Function],
                      ";color:",
                      [Function],
                      ";background-color:",
                      [Function],
                      ";overflow:hidden;transition:0.6s;",
                      "@media (min-width:",
                      "35.5",
                      "em){",
                      "
    display: block;
  ",
                      "}",
                      ";",
                    ],
                  },
                  "displayName": "styles__Content",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Content-xptth5-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="styles__Content-xptth5-0 iheHWk"
              >
                <_default>
                  <div>
                    Some things I've been working on.
                  </div>
                </_default>
              </div>
            </StyledComponent>
          </styles__Content>
        </Memo()>
      </div>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;
